time/src/main/java/net/jqwik/time/internal/properties/arbitraries/DefaultPeriodArbitrary.java

```diff
@@ -15,21 +16,27 @@
 
 	private int yearsMin = Integer.MIN_VALUE;
 	private int yearsMax = Integer.MAX_VALUE;
+	private BigInteger yearPeriod = null;
 
 	private int monthsMin = 0;
 	private int monthsMax = 11;
+	private BigInteger monthPeriod = null;
 
 	private int daysMin = 0;
 	private int daysMax = 30;
+	private BigInteger dayPeriod = null;
+
+	private BigInteger helperMonthDay = null;
 
 	@Override
 	protected Arbitrary<Period> arbitrary() {
 
-		IntegerArbitrary years = Arbitraries.integers().between(yearsMin, yearsMax);
-		IntegerArbitrary months = Arbitraries.integers().between(monthsMin, monthsMax);
-		IntegerArbitrary days = Arbitraries.integers().between(daysMin, daysMax);
+		BigInteger bigIntegerStart = BigInteger.ZERO;
+		BigInteger bigIntegerEnd = calculateBigIntegerEnd();
+
+		BigIntegerArbitrary numbers = Arbitraries.bigIntegers().between(bigIntegerStart, bigIntegerEnd);
 
-		Arbitrary<Period> periodArbitrary = Combinators.combine(years, months, days).as(Period::of);
+		Arbitrary<Period> periodArbitrary = numbers.map(this::calculatePeriod);
```

Comment:
Here is another solution. Shrinking works, but equal distribution fails (positive and negative)

```
timestamp = 2021-01-22T13:01:25.106151200, [CheckEqualDistribution:periodCanBePositiveAndNegative] (1000) Period is negative = 
    true  (953) : 95 %
    false ( 47) :  5 %
```

Which solution is better?

Reply:
`Arbitrary<BigInteger> numbers = Arbitraries.bigIntegers().withDistribution(RandomDistribution.uniform()).between(bigIntegerStart, bigIntegerEnd);`

Found a solution :)